File: day5/puzzle1.js
var fs = require('fs');

function count(re, s) {
 for (var i = 0; m = re.exec(s); ++i);
 return i;
}

fs.readFile('day5.txt', 'utf-8', function (err, inp) {
 if (err) throw err;
 console.log(inp.trim().split(/\n/g).reduce(function (curr, line) {
  var ok = (count(/[aeiou]/g,line) >= 3 && count(/(.)\1/g,line) > 0 && count(/(ab|cd|pq|xy)/g,line) == 0)
  return curr + (ok ? 1 : 0);
 }, 0));
});

File: 2015/05/puzzle1.py
import re
c = 0
for i in open('day5.txt'):
 m1 = len(re.findall("[aeiou]", i))
 m2 = len(re.findall(r"(.)\1", i))
 m3 = len(re.findall("(ab|cd|pq|xy)", i))
 m = (m1 >= 3) and (m2 > 0) and (m3 == 0)
 if m:
  c = c + 1
print c
